src/main/kotlin/org/rust/ide/intentions/UnwrapLambdaExprIntention.kt

```diff
@@ -0,0 +1,39 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Condition
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.RustBlockExprElement
+import org.rust.lang.core.psi.RustElementFactory
+import org.rust.lang.core.psi.RustLambdaExprElement
+import org.rust.lang.core.psi.util.parentOfType
+
+class UnwrapLambdaExprIntention : PsiElementBaseIntentionAction() {
+    override fun getText() = "Remove braces from lambda expression"
+    override fun getFamilyName() = getText()
+    override fun startInWriteAction() = true
+
+    override fun invoke(project: Project, editor: Editor, element: PsiElement) {
+        val lambdaExpr = element.parentOfType<RustLambdaExprElement>() ?: return
+        val block = PsiTreeUtil.findFirstParent(element, blockOrLambda) as? RustBlockExprElement ?: return
+        val paramsStr = lambdaExpr.parameters.text
+        val bodyStr = block.block?.expr?.text ?: return
+        val unbraced = RustElementFactory.createUnbracedLambdaExpr(project, paramsStr, bodyStr) ?: return
+        lambdaExpr.replace(unbraced)
+    }
+
+    override fun isAvailable(project: Project, editor: Editor, element: PsiElement): Boolean {
+        if (!element.isWritable) return false
+        val block = PsiTreeUtil.findFirstParent(element, blockOrLambda) as? RustBlockExprElement ?: return false
+        if (block.parent !is RustLambdaExprElement) return false
```

Comment:
@mkaput this is a little bit confusing that if we want block, why should we find block or lambda first?
```
// find first parent which is either block or lambda
// if we get lambda, than return false as we are inside "unblocked" (hehe) lambda
```